### PR TITLE
qtum contractjs

### PR DESCRIPTION
**Pull request by patidarmanoj10**

### CODE REVIEW

**Comment by pablen** [comment]
@patidarmanoj10 a few nitpicks
- can you remove Node 6 in `.travis.yml` so that the build checks pass again? it seems that a new dependency is not compatible with Node 6
- there is a reference to `morden` in `README.md`, could you update it with Mordor?

Everything else LGTM.

**Comment by patidarmanoj10** [comment]
@pablen  looks good now?

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
+  - "11"
```
Comment (gabmontes) [inline .travis.yml]: Let's test it in 12, which will be LTS. 11 is an intermediate version. Not worth testing contracts in that one.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "metronome-contracts",
-  "version": "2.4.0",
+  "version": "2.4.2",
```
Comment (gabmontes) [inline package.json]: Don't change the version now. We will publish once this PR is merged and will update the version in a new PR.

File: src/contracts.json [hunk 3]
```diff
@@ -50,32 +50,29 @@
     },
     "version": "35a6cd3"
   },
-  "morden": {
-    "birthblock": 3757156,
+  "mordor": {
+    "birthblock": 0,
```
Comment (gabmontes) [inline src/contracts.json]: Can we set it to the proper height of the tx(s) that created the contracts?

File: src/contracts.json [hunk 4]
```diff
@@ -50,32 +50,29 @@
...
+        "address": "0x62187b4AA3BDA5bF7580fa4aa89659f018bb958f"
       },
       "Validator": {
-        "address": "0xdb71855ae18BB34497B49Fe673fA9D738B397Fe7",
-        "version": "18e06e7"
+        "address": "0x444E8fafB172cF0Df4Be1B4c20e24a48bd4a0Ae0"
       }
     },
-    "version": "4af63f2"
+    "version": "mordorv1"
```
Comment (gabmontes) [inline src/contracts.json]: Let's use the actual commit hash of the metronome repo used to build the contracts deployed.

File: src/contracts.json [hunk 5]
```diff
@@ -103,5 +100,29 @@
       }
     },
     "version": "35a6cd3"
+  },
+  "qtumTestnet": {
+    "birthblock": 222408,
+    "version": "qtumv1",
```
Comment (gabmontes) [inline src/contracts.json]: Same here. Let's use the actual commit hash.

File: test/index.test.js [hunk 6]
```diff
@@ -2,10 +2,12 @@
 
 const Web3 = require('web3')
 const web3 = new Web3('http://localhost:8545')
+const qtumrpc = 'http://qtum:test@ec2-54-172-219-152.compute-1.amazonaws.com:3889'
```
Comment (gabmontes) [inline test/index.test.js]: Let's not expose internal testing URLs. All defaults in the code should be set to work 100% in the local dev env (i.e. `localhost`).

File: src/contracts.json [hunk 7]
```diff
@@ -103,5 +100,29 @@
...
+    "contracts": {
+      "Auctions": {
+        "address": "b3f2d768fb1f2e341f5fe72f1b207fada4e51fe3"
```
Comment (gabmontes) [inline src/contracts.json]: Let's use checksum address whenever possible.

File: test/index.test.js [hunk 8]
```diff
@@ -31,6 +33,26 @@ const contracts = {
     METToken: {
       address: '0xF3e9a687Fdf24112745D4d7dEe150BA87A07ecc3'
     }
+  },
+  qtumTestnet: {
+    Auctions: {
+      address: 'b3f2d768fb1f2e341f5fe72f1b207fada4e51fe3'
```
Comment (gabmontes) [inline test/index.test.js]: Let's use checksum address here too.

File: .travis.yml (hunk 1 above)
Comment (patidarmanoj10) [reply .travis.yml]: Done

File: package.json (hunk 2 above)
Comment (patidarmanoj10) [reply package.json]: done

File: src/contracts.json (hunk 4 above)
Comment (patidarmanoj10) [reply src/contracts.json]: done.

File: test/index.test.js (hunk 6 above)
Comment (patidarmanoj10) [reply test/index.test.js]: done

File: src/contracts.json (hunk 3 above)
Comment (patidarmanoj10) [reply src/contracts.json]: this was set intentionally as Mordor just new chain and contract deployed is around 35000 block. Anyway. I have set the exact number

**Pull request merged**